scipy/spatial/distance.py

```diff
@@ -145,15 +145,13 @@ def _convert_to_double(X):
         X = X.copy()
     return X
 
+def _vdot(a, b):
```

Comment:
No, because inner has output shape `out.shape = a.shape[:-1] + b.shape[:-1]`, whereas this has `out.shape = np.broadcast(a, b)[0].shape[:-1]`

Ie, `np.inner(np.zeros((1, 2, 3)), np.zeros((4, 1, 3))).shape == (1, 2, 4, 1)`, whereas this results in `(4, 2)`